roles/load_aware_deploy_trimaran/tasks/main.yml

```diff
@@ -12,28 +12,55 @@
   delay: 3
   retries: 20
   # until all of the pods are in the running state
-  until: "'Completed' not in monitoring_enabled.stdout and 'Failed' not in monitoring_enabled.stdout and 'Pending' not in monitoring_enabled.stdout" 
+  until: "'Completed' not in monitoring_enabled.stdout and 'Failed' not in monitoring_enabled.stdout and 'Pending' not in monitoring_enabled.stdout"
 
-- name: Deploy Trimaran scheduler
+- name: Get Monitoring Secret
   shell:
+    oc get secret -n openshift-user-workload-monitoring
+      | grep  prometheus-user-workload-token
+      | head -n 1
+      | awk '{print $1 }'
+  register: monitoring_secret
+  delay: 3
+  retries: 20
+  until: "'prometheus-user-workload-token' in monitoring_secret.stdout"
+
```

Comment:
Here you can add a task like:
```
- name: Wait for the secret to look consistent
  shell:
   set -o pipefail;
    oc get secret $MONITORING_SECRET -n openshift-user-workload-monitoring -ojson 
        | jq -r '.data.token' | base64 -d | wc -c 
  register: secret_token_length
  delay: 3
  retries: 20
  until: secret_token_length.stdout | int > 1000
```
(test it locally first, I'm not 100% of the code)